### PR TITLE
[Docs] Improve VLM docs

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR improve the docs of various VLMs:

* llava: batched inference
* llava-onevision: minor nit on model naming
* qwen2-vl: add to appropriate multimodal section in toctree, add figure
* pixtral: add `apply_chat_template`, add figure

### CODE REVIEW

File: docs/source/en/model_doc/qwen2_vl.md [hunk 1]
```diff
@@ -217,28 +217,30 @@ print(output_text)
 
 ### Usage Tips
 
-#### Image Resolution for performance boost
+#### Image Resolution trade-off
 
-The model supports a wide range of resolution inputs. By default, it uses the native resolution for input, but higher resolutions can enhance performance at the cost of more computation. Users can set the minimum and maximum number of pixels to achieve an optimal configuration for their needs.
```
Comment (amyeroberts) [inline docs/source/en/model_doc/qwen2_vl.md:222]: The previous comment was better. Perhaps we can introduce the example below after the initial introduction to something like `"In the case of limited GPU RAM, one can reduce the resolution, as follows:"`

File: docs/source/en/model_doc/qwen2_vl.md [hunk 2]
```diff
@@ -217,28 +217,30 @@ print(output_text)
...
+The model supports a wide range of resolution inputs. By default, it uses the native resolution for input, but one can reduce the resolution in case of limited GPU RAM, as follows:
 
 ```python
+min_pixels = 256*28*28
+max_pixels = 1024*28*28 
+processor = AutoProcessor.from_pretrained("Qwen/Qwen2-VL-7B-Instruct", min_pixels=min_pixels, max_pixels=max_pixels)
+```
+This ensures each image gets encoded using a number between 256-1024 tokens.
```
Comment (amyeroberts) [inline docs/source/en/model_doc/qwen2_vl.md:229]: It would be good to explain where the 28 comes in here

File: docs/source/en/model_doc/llava.md [hunk 3]
```diff
@@ -75,6 +77,60 @@ print(text_prompt)
...
+image_cats = Image.open(requests.get(url, stream=True).raw)
+
+# Prepare a batch of two prompts
+conversation_1 = [
+    {
+        "role": "user",
+        "content": [
+            {"type": "image"},
+            {"type": "text", "text": "What is shown in this image?"},
+            ],
```
Comment (amyeroberts) [inline docs/source/en/model_doc/llava.md:108]:
```suggestion
        ],
```

File: docs/source/en/model_doc/llava.md [hunk 4]
```diff
@@ -75,6 +77,60 @@ print(text_prompt)
...
+    },
+]
+
+conversation_2 = [
+    {
+        "role": "user",
+        "content": [
+            {"type": "image"},
+            {"type": "text", "text": "What is shown in this image?"},
+            ],
```
Comment (amyeroberts) [inline docs/source/en/model_doc/llava.md:118]:
```suggestion
        ],
```

**Review by amyeroberts — COMMENTED** [review]
Thanks for adding these updates! 

Just some small nits

cc @zucchini-nlp as this involves VLMs

File: docs/source/en/model_doc/qwen2_vl.md (hunk 1 above)
Comment (NielsRogge) [reply docs/source/en/model_doc/qwen2_vl.md:222]: Done

File: docs/source/en/model_doc/llava.md [hunk 5]
```diff
@@ -40,7 +40,9 @@ The original code can be found [here](https://github.com/haotian-liu/LLaVA/tree/
 
 - Note the model has not been explicitly trained to process multiple images in the same prompt, although this is technically possible, you may experience inaccurate results.
 
-- For better results, we recommend users to use the processor's `apply_chat_template()` method to format your prompt correctly. For that you need to construct a conversation history, passing in a plain string will not format your prompt. Each message in the conversation history for chat templates is a dictionary with keys "role" and "content". The "content" should be a list of dictionaries, for "text" and "image" modalities, as follows:
+### Single image inference
+
+For best results, we recommend users to use the processor's `apply_chat_template()` method to format your prompt correctly. For that you need to construct a conversation history, passing in a plain string will not format your prompt. Each message in the conversation history for chat templates is a dictionary with keys "role" and "content". The "content" should be a list of dictionaries, for "text" and "image" modalities, as follows:
```
Comment (amyeroberts) [inline docs/source/en/model_doc/llava.md:45]: Could we link to a page about chat templates here instead? We probably don't want to have to repeat info about the structure in all the model pages

File: docs/source/en/model_doc/pixtral.md [hunk 6]
```diff
@@ -18,69 +18,62 @@ rendered properly in your Markdown viewer.
...
 ]
-PROMPT = "<s>[INST]Describe the images.\n[IMG][IMG][IMG][IMG][/INST]"
 
-inputs = processor(images=IMG_URLS, text=PROMPT, return_tensors="pt").to("cuda")
+prompt = processor.apply_chat_template(chat)
+inputs = processor(text=prompt, images=[url_dog, url_mountain], return_tensors="pt").to(model.device)
 generate_ids = model.generate(**inputs, max_new_tokens=500)
 output = processor.batch_decode(generate_ids, skip_special_tokens=True, clean_up_tokenization_spaces=False)[0]
-
-EXPECTED_GENERATION = """
```
Comment (amyeroberts) [inline docs/source/en/model_doc/pixtral.md:60]: Why remove the expected generation?

**Review by amyeroberts — APPROVED** [review]
Thanks for adding these improvements!

**Pull request merged**